clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/provider/view/KubernetesV2JobProvider.java

```diff
@@ -86,7 +89,30 @@ public KubernetesV2JobStatus collectJob(String account, String location, String
   }
 
   public Map<String, Object> getFileContents(String account, String location, String id, String filename) {
```

Comment:
This function looks almost identical to `collectJob` above; is there a way to factor out the common logic?